### ISSUE TITLE
Error returns from postUpdateContext

### ISSUE THREAD

**Issue by kzangeli** [telefonicaid/fiware-orion#2763]
Right now it seems we have no less than **three** ways to return errors in the service routine  postUpdateContext.

See postEntities.cpp, after the call to postUpdateContext:
* ciP->httpStatusCode
* parseDataP->upcrs.res.oe.code != SccNone
* parseDataP->upcrs.res.errorCode.code != SccNone (or similar)

It would be good to have **one** way of detecting errors, not three ...

**Comment by fgalan** [comment]
I have set P6 for this, but change if if you prefer other @kzangeli

**Comment by fgalan** [comment]
(Moving from 1.7.0 milestone to Refactoring)

**Comment by kzangeli** [comment]
Another bug that is probably related:

See functest `2750_incoming_errors.test`, step 05:

    orionCurl --url /version --servicePath XXX

This request returns:

    HTTP/1.1 200 OK
    Content-Length: 111
    Content-Type: application/json
    Fiware-Correlator: REGEX([0-9a-f\-]{36})
    Date: REGEX(.*)

    {
      "description": "Only /absolute/ Service Paths allowed [a service path must begin with /]",
      "error": "BadRequest"
    }

Note the **200 OK**
Should be **400 Bad Request**

**Comment by kzangeli** [comment]
The `200 OK` in Step 05 of `2750_incoming_errors.test` seems to have been fixed. It now gives a `400 Bad Request` (Jan 2017 in master).

**Comment by kzangeli** [comment]
The error in 2750_incoming_errors.test is **not** related to this issue.
What is happening there is that /version is not API v2, but it behaves a little like API v1 that always returns `200 OK`

**Comment by fgalan** [comment]
After PR https://github.com/telefonicaid/fiware-orion/pull/2821, I understand that the "bufixing part" of this issue is completed (for the records: this happends during 1.7.0 development cycle). However, I understand that the issue have to keep opened for the "refactor parts" (and, with that regards, it is correctly clasified in the Refactor milestone).

Doubt: once the bugfixing part is over, is the bug/2763_postUpdateContext_error_return needed? Or should be removed?

**Comment by kzangeli** [comment]
The branch can be removed.   The issue must stay as it is really not about this bug, but about refactoring

**Comment by fgalan** [comment]
bug/2763_postUpdateContext_error_return branch deleted

**Comment by fgalan** [comment]
(PR #2821 is referred here, as there was a bug related with this issue that was fixed in that PR)

**Comment by Anjali-NEC** [comment]
Hi @fgalan @kzangeli 
I would like to work on this issue, please let me know this issue is still valid or not.

**Comment by fgalan** [comment]
The issue is still valid but before starting to implement it I think it needs a analysis phase to identify what to be exactly done. Thus maybe its better you choose another issue with a more straightforward task to do.

**Comment by fgalan** [comment]
After 4125c4d4032fe56f3575e4798193040279aa4cc6 in PR (maybe even before?) there is only one way of returning in the service routine postUpdateContext, which is `parseDataP->upcrs.res.oe.code != SccNone`

For instance in postEntities() routine now we have:

```
  //
  // 03. Check error
  //
  std::string  answer = "";
  if (parseDataP->upcrs.res.error.code != SccOk)
  {
    ciP->httpStatusCode = parseDataP->upcrs.res.error.code;
    TIMED_RENDER(answer = parseDataP->upcrs.res.error.toJson());
    ciP->answer         = answer;
  }
  else
  {
  ...
  }
```

Thus this issue can be considered to be solved.
